src/game_renderer/GameRenderer.cpp

```diff
@@ -133,12 +135,17 @@ void GameRenderer::Render(std::shared_ptr<GameState> game_state) {
   glUniformMatrix4fv(current_program->getUniform("V"), 1, GL_FALSE,
                      glm::value_ptr(V.topMatrix()));
 
-  for (std::shared_ptr<Platform> platform : *level->getPlatforms()) {
-    if (!ViewFrustumCulling::IsCulled(platform->GetBoundingBox(), vfplane)) {
-      MV = platform->GetTransform();
-      glUniformMatrix4fv(current_program->getUniform("MV"), 1, GL_FALSE,
-                         glm::value_ptr(MV.topMatrix()));
-      platform->GetModel()->draw(current_program);
+  for (std::shared_ptr<GameObject> obj : *level->getObjects()) {
+    if (obj->isObstacle()) {
+      if (Platform* platform = dynamic_cast<Platform*>(obj.get())) {
```

Comment:
needs to be std::dynamic_pointer_cast because here we need the specific type of the object (Platform in this case) to determine rendering information. That's also why we can't combine the rendering loops. because we need to bind a different program of each type of object. We already have culling in place so looping though all the objects in view a couple times during the render phase shouldn't be a problem
Alex is also looking into speeding up the drawing process
https://github.com/bbeck13/RhythmRunner/issues/58